lib/internal/child_process.js

```diff
@@ -189,8 +189,8 @@ function ChildProcess() {
     //
     // - spawn failures are reported with exitCode < 0
     //
-    var syscall = self.spawnfile ? 'spawn ' + self.spawnfile : 'spawn';
-    var err = (exitCode < 0) ? errnoException(exitCode, syscall) : null;
+    const syscall = self.spawnfile ? 'spawn ' + self.spawnfile : 'spawn';
```

Comment:
Possibly, I found that it is in some ternary uses and not in others, so I'm not sure exactly what triggers it. Executing this code path just needs to be checked with `--trace-opt --trace-deopt` to be sure.